src/GraphQL.Tests/Federation/InterfaceEntityTests.cs

```diff
@@ -0,0 +1,282 @@
+using GraphQL.Federation;
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Tests.Federation;
+
+public class InterfaceEntityTests
```

Comment:
Schema-first and Code-first tests were added to ensure it works properly.

Reply:
And type-first, which required even more changes (but not more breaking changes)